core/src/com/unciv/logic/map/tile/TileStatFunctions.kt

```diff
@@ -76,14 +59,16 @@ class TileStatFunctions(val tile: Tile) {
 
             if (stats.gold != 0f && observingCiv.goldenAges.isGoldenAge())
                 stats.gold++
-        }
-        if (tile.isCityCenter()) {
-            if (stats.food < 2) stats.food = 2f
-            if (stats.production < 1) stats.production = 1f
+
+            if (improvement != null) {
+                val ensureMinUnique = improvement
+                    .getMatchingUniques(UniqueType.EnsureMinimumStats, stateForConditionals)
+                    .firstOrNull()
+                if (ensureMinUnique != null) minimumStats = ensureMinUnique.stats
+            }
         }
 
-        for ((stat, value) in stats)
-            if (value < 0f) stats[stat] = 0f
+        stats.coerceAtLeast(minimumStats)  // Minimum 0 or as defined by City center
```

Comment:
Since this is the only place this is actually used, I don't think we need a special function for that. We can do it here, looping over key/value pairs in the stats, in 3 lines of code. "for stat,value in stats.hashmap(); minimumValue = minimumStats[stat]; if value<minimumValue stats[stat]=minimumValue"

Reply:
I would still go with B 
If we see a third usage then I would go with C

Reply:
"Three's a crowd"

Reply:
You're right, got mixed up